src/mlpack/methods/ann/layer/transposed_convolution_impl.hpp

```diff
@@ -77,25 +82,134 @@ TransposedConvolution<
   weights.set_size((outSize * inSize * kernelWidth * kernelHeight) + outSize,
       1);
 
-  aW = (outputWidth + 2 * padWidth - kernelWidth) % strideWidth;
-  aH = (outputHeight + 2 * padHeight - kernelHeight) % strideHeight;
+  // Transform paddingType to lowercase.
+  std::string paddingTypeLow = paddingType;
+  std::transform(paddingType.begin(), paddingType.end(), paddingTypeLow.begin(),
+      [](unsigned char c){ return std::tolower(c); });
+  if (paddingTypeLow == "valid")
+  {
+    // Set Padding to 0.
+    padWLeft = 0;
+    padWRight = 0;
+    padHTop = 0;
+    padHBottom = 0;
+  }
+  else if (paddingTypeLow == "same")
+  {
+    InitializeSamePadding();
+  }
+  size_t totalPadWidth = padWLeft + padWRight;
+  size_t totalPadHeight = padHTop + padHBottom;
+  aW = (outputWidth + totalPadWidth - kernelWidth) % strideWidth;
+  aH = (outputHeight + totalPadHeight - kernelHeight) % strideHeight;
+
+  const size_t padWidthLeftForward = kernelWidth - padWLeft - 1;
+  const size_t padHeightTopForward = kernelHeight - padHTop - 1;
+  const size_t padWidthRightForward = kernelWidth - padWRight - 1;
+  const size_t padHeightBottomtForward = kernelHeight - padHBottom - 1;
+
+  paddingForward = ann::Padding<>(padWidthLeftForward,
+      padWidthRightForward + aW, padHeightTopForward,
+      padHeightBottomtForward + aH);
+  paddingBackward = ann::Padding<>(padWLeft, padWRight, padHTop, padHBottom);
+
+  // Check if the output height and width are possible given the other
+  // parameters of the layer.
+  if (outputWidth != strideWidth * (inputWidth - 1) +
+              aW + kernelWidth - totalPadWidth ||
```

Comment:
The lines should be tabbed twice, see https://github.com/mlpack/mlpack/wiki/DesignGuidelines#line-length-and-wrapping for more details.

Reply:
I have made the changes. Have a look when you get the chance.